### DISCUSSION TITLE
DDPP 第二章学习笔记

### DISCUSSION THREAD

**Discussion by giscus** [ouuan/iles-blog#24] [Comments]
# DDPP 第二章学习笔记

DDPP5 第二章“Number Systems and Codes”的学习笔记。

https://ouuan.moe/post/2023/01/ddpp-2

**Comment by ouuan** [comment]
学完去看了眼不知道是不是教材的《数字逻辑与数字集成电路（第2版 ）》的目录，第一章先是 2、8、16 进制，然后是整数编码和浮点数编码，还有个十进制数编码，最后是 ASCII，这些我都跳过了，写了笔记的就只有一个格雷码，笔记里剩下的东西这个书上都没有。而且这书还把 ones’ complement 写成 one’s complement，果然不太靠谱（

**Reply by YfLi03** [reply]
two's complement 但是 ones' complement

还真是容易混淆（

**Reply by ouuan** [reply]
知道为什么叫这个名字就很好记了：（CS:APP3e 北美版 p68，中文版 p48）

![图片](https://user-images.githubusercontent.com/30581822/214458292-64dbdb1a-9ad0-47a7-8176-dc3833f1eec5.png)

所以写错说明这名字是硬背的（

**Reply by YfLi03** [reply]
学到许多 (看来 csapp 还是得好好读）
